### PR TITLE
Fix excessive warning

### PR DESCRIPTION
**Pull request by lubennikovaav**

### CODE REVIEW

File: pageserver/src/layered_repository/inmemory_layer.rs [hunk 1]
```diff
@@ -369,10 +370,17 @@ impl InMemoryLayer {
 
         if old.is_some() {
             // We already had an entry for this LSN. That's odd..
-            warn!(
-                "Page version of rel {} blk {} at {} already exists",
-                self.seg.rel, blknum, lsn
-            );
+            // That is fine for checkpoint record that we update
+            // at the end of timeline WAL import.
+            match self.seg.rel {
```
Comment (LizardWizzard) [inline pageserver/src/layered_repository/inmemory_layer.rs:375]: nit: can be
```rust
if !matches!(self.seg.rel, RelishTag::Checkpoint) {
 ...
}
```
maybe even add it to `if old.is_some()`

Comment (lubennikovaav) [reply pageserver/src/layered_repository/inmemory_layer.rs]: > nit: can be
> 
> ```rust
> if !matches!(self.seg.rel, RelishTag::Checkpoint) {
>  ...
> }
> ```
> 
> maybe even add it to `if old.is_some()`

Thank you! This syntax is way better.

**Comment by hlinnaka** [comment]
Are you still seeing the warnings after commit ab33614ab166d53d0a35837f2e9103f041378f9a?

**Comment by lubennikovaav** [comment]
> Are you still seeing the warnings after commit [ab33614](https://github.com/zenithdb/zenith/commit/ab33614ab166d53d0a35837f2e9103f041378f9a)?

Nope. I found this warning in the test log, but it must have been leftovers from some very old test tuns.

**Pull request closed without merging**